test/e2e/downwardapi_volume.go

```diff
@@ -18,57 +18,180 @@ package e2e
 
 import (
 	"fmt"
+	"strings"
+	"time"
 
 	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/util"
 
 	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
 )
 
+// How long to wait for a log pod to be displayed
+var podLogTimeout = 15 * time.Second
+
 var _ = Describe("Downward API volume", func() {
 	f := NewFramework("downward-api")
-
-	It("should provide labels and annotations files [Conformance]", func() {
+	It("should provide podname only [Conformance]", func() {
 		podName := "downwardapi-volume-" + string(util.NewUUID())
-		pod := downwardAPIVolumePod(podName)
+		pod := downwardAPIVolumePod(podName, map[string]string{}, map[string]string{}, "/etc/podname")
 
 		testContainerOutput("downward API volume plugin", f.Client, pod, 0, []string{
-			fmt.Sprintf("cluster=\"rack10\"\n"),
-			fmt.Sprintf("builder=\"john-doe\"\n"),
 			fmt.Sprintf("%s\n", podName),
 		}, f.Namespace.Name)
 	})
 
-	It("should provide labels and annotations files as non-root with fsgroup [Conformance] [Skipped]", func() {
+	It("should provide podname as non-root with fsgroup [Conformance] [Skipped]", func() {
 		podName := "metadata-volume-" + string(util.NewUUID())
 		uid := int64(1001)
 		gid := int64(1234)
-		pod := downwardAPIVolumePod(podName)
+		pod := downwardAPIVolumePod(podName, map[string]string{}, map[string]string{}, "/etc/podname")
 		pod.Spec.SecurityContext = &api.PodSecurityContext{
 			RunAsUser: &uid,
 			FSGroup:   &gid,
 		}
 		testContainerOutput("downward API volume plugin", f.Client, pod, 0, []string{
-			fmt.Sprintf("cluster=\"rack10\"\n"),
-			fmt.Sprintf("builder=\"john-doe\"\n"),
 			fmt.Sprintf("%s\n", podName),
 		}, f.Namespace.Name)
 	})
+
+	It("should verify update on labels modification [Conformance]", func() {
```

Comment:
minor nit: the test case name describes the functionality you're testing, not what you're doing... so I think maybe this could be shortened to something like "should update labels on modification"